### PR TITLE
Mention Piwik 2 is no longer supported (Piwik 2.18)

### PR DESCRIPTION
**Pull request by tsteur**
Tried to keep it simple... showing warning to all admins/super users in reporting/admin UI:

image

And simple page showing more info
apache matomo_index php_module corehome action piwikisoutdated idsite 1 period day date yesterday

didn't add any of the piwik is now matomo image etc as didn't want to put it under GPL and I would say it is better to keep it straight to the point.

piwikisoutdated

refs https://github.com/matomo-org/matomo/issues/11810

### CODE REVIEW

**Comment by tsteur** [comment]
Changed it to at least view access. Would in theory need to show super user email login names or their email so they can ask them to update.

File: core/Plugin/ControllerAdmin.php [hunk 1]
```diff
@@ -213,6 +215,43 @@ private static function isUsingPhpVersionCompatibleWithNextPiwik()
         return version_compare( PHP_VERSION, self::getNextRequiredMinimumPHP(), '>=' );
     }
 
+    public static function notifyPiwik2EndOfLife()
+    {
+        if (!Piwik::isUserHasSomeViewAccess()) {
+            return;
+        }
```
Comment (diosmosis) [inline core/Plugin/ControllerAdmin.php]: Should this show for anonymous users?

File: plugins/CoreHome/templates/piwikIsOutdated.twig [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                A user with access to the server can edit the <code>config/config.ini.php</code> file and add the following lines (the line with <code>[General]</code> may already exist):
+                <br />
+                <br />
+                <pre><code>[General]
+show_piwik2_end_of_life_warning = 0</code></pre>
+            </p>
+            {% endif %}
+        </div>
+    </div>
+{% endblock %}
```
Comment (diosmosis) [inline plugins/CoreHome/templates/piwikIsOutdated.twig]: I guess this shouldn't be translated?

File: plugins/CoreHome/Controller.php [hunk 3]
```diff
@@ -49,6 +49,13 @@ public function getDefaultAction()
         return 'redirectToCoreHomeIndex';
     }
 
+    public function piwikIsOutdated()
+    {
+        Piwik::checkUserHasSomeViewAccess();
```
Comment (diosmosis) [inline plugins/CoreHome/Controller.php]: Should this be shown to anonymous viewers?

File: plugins/CoreHome/templates/piwikIsOutdated.twig (hunk 2 above)
Comment (tsteur) [reply plugins/CoreHome/templates/piwikIsOutdated.twig]: No translation needed.  Discussed it in the chat a while ago. It wouldn't be translated in time likely anyway and it'll be the last update.

File: core/Plugin/ControllerAdmin.php (hunk 1 above)
Comment (tsteur) [reply core/Plugin/ControllerAdmin.php]: I would say so. I think we discussed it in the chat but not 100% sure. Maybe @mattab can confirm.

**Pull request merged**